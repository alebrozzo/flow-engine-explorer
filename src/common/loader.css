.loader {
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
}

.loader .loaderBackdrop {
    display: flex;
    align-items: center;
    justify-content: center;
    height: 100vh;
    background: rgba(255, 255, 255, 0.8);
}

.loader .spinner {
}
