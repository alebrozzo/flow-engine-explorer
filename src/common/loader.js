import * as React from "react";

import { Spinner } from "./spinner";
import "./loader.css";

export const Loader = () => (
    <div className="loader">
        <div className="loaderBackdrop">
            <div className="spinner">
                <Spinner />
            </div>
        </div>
    </div>
);
