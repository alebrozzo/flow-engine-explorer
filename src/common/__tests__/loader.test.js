import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

import { Loader } from "../loader";

describe("Loader", () => {
    it("renders a loader", () => {
        const component = shallow(<Loader />);
        expect(toJson(component)).toMatchSnapshot();
    });
});
