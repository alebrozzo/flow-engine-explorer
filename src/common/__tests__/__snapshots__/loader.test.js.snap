// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader renders a loader 1`] = `
<div
  className="loader"
>
  <div
    className="loaderBackdrop"
  >
    <div
      className="spinner"
    >
      <Spinner />
    </div>
  </div>
</div>
`;
